### DISCUSSION TITLE
Using the additive manufacturing plugin (AM Modeler)

### DISCUSSION THREAD

**Discussion by lukashof** [haiiliin/abqpy#4870] [Ideas]
Hey all 👋,

thank you, @haiiliin, for this package, it really goes a long way to make working with ABAQUS more enjoyable!

I am working with the [AM Modeler plugin](https://help.3ds.com/2023/english/dssimulia_established/SIMACAECAERefMap/simacae-m-PlgExampleAdditiveManufacturing-sb.htm?contextscope=all&id=11d9b1ea8ac543b685510222d0095205) and found that abqpy doesn't provide hints for it yet. I was wondering whether it would be feasible to add support for the plugin.

In the .rpy created by ABAQUS CAE I found that it has to be imported like this:
![grafik](https://github.com/haiiliin/abqpy/assets/100953722/a347f988-02f0-4062-a7b4-ea11b24c7056)

I am not sure whether you created this plugin by manually searching all available Python commands or if there is a simpler way that allows you to add the hints more easily. I would be really happy if you let me know what you think!

**Comment by haiiliin** [comment]
The library was created by crawling the [Abaqus Scripting Reference](https://help.3ds.com/2023/english/dssimulia_established/SIMACAEKERRefMap/simaker-m-PythonCommands-sb.htm?contextscope=all) documentation and reconstructing the documents into Python typing annotations. Since the APIs of `AM Modeler plugin` are not documented in the Abaqus Scripting Reference, it would be not easy to rebuild the Python typing annotations.

**Reply by lukashof** [reply]
Thanks for the quick reply!
Okay, that's a bummer 😕, but thanks for letting me know! I don't think I have the patience at the moment to manually find the commands and write annotations for them.
